dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ProcessingManager.java

```diff
@@ -0,0 +1,75 @@
+package com.dat3m.dartagnan.program.processing;
+
+import com.dat3m.dartagnan.program.Program;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.sosy_lab.common.configuration.Configuration;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+import org.sosy_lab.common.configuration.Option;
+import org.sosy_lab.common.configuration.Options;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.dat3m.dartagnan.configuration.OptionNames.*;
+
+@Options
+public class ProcessingManager implements ProgramProcessor {
+
+    private final static Logger logger = LogManager.getLogger(ProcessingManager.class);
+
+    private final List<ProgramProcessor> programProcessors = new ArrayList<>();
+
+    // =========================== Configurables ===========================
+
+    @Option(name = REDUCE_SYMMETRY,
+            description = "Reduces the symmetry of the program (unsound in general).",
+            secure = true)
+    private boolean reduceSymmetry = false;
+
+	@Option(name= ATOMIC_BLOCKS_AS_LOCKS,
+			description="Transforms atomic blocks by adding global locks.",
+			secure=true)
+		private boolean atomicBlocksAsLocks = false;
+
+	@Option(name= CONSTANT_PROPAGATION,
+			description="Perform constant propagation.",
+			secure=true)
+		private boolean constantPropagation = false;
+
+    // ======================================================================
+
+    private ProcessingManager(Configuration config) throws InvalidConfigurationException {
+        config.inject(this);
+
+        programProcessors.addAll(Arrays.asList(
+                DeadCodeElimination.fromConfig(config),
+                BranchReordering.fromConfig(config),
+                Simplifier.fromConfig(config),
+                LoopUnrolling.fromConfig(config)
+        ));
+        if(constantPropagation) {
+            programProcessors.add(ConstantPropagation.fromConfig(config));
+        }
+        programProcessors.add(Compilation.fromConfig(config));
+        if(atomicBlocksAsLocks) {
+            // TODO: Do we really want to execute this after compilation?
+            programProcessors.add(AtomicAsLock.fromConfig(config));
```

Comment:
I propose moving `AtomicAsLock` before `Compilation`, as it generates events that should get individual `cId`s.